wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php

```diff
@@ -1669,11 +1669,14 @@ public static function generateInvoiceNumber(string $op, string $objectName, int
       return;
     }
 
+    $lock = \Civi::lockManager()->acquire('invoice_number_generation');
+    if (!$lock->isAcquired()) {
+      \Civi::log()->warning('Could not acquire lock for invoice number generation. Skipping for contribution ID: ' . $objectRef->id);
+      return;
+    }
+
     try {
       $contributionId = $objectRef->id;
-      if (!$contributionId) {
-        return;
-      }
```

Comment:
This was removed because this was already being ensured earlier in the method:
```php
if ($objectName !== 'Contribution' || !$objectRef->id) {
  return;
}
```